src/modules/connections-tree/items/abstractnamespaceitem.cpp

```diff
@@ -28,6 +28,16 @@ QSharedPointer<TreeItem> AbstractNamespaceItem::child(uint row) const {
   return QSharedPointer<TreeItem>();
 }
 
+QRegExp AbstractNamespaceItem::getFilter() const {
+  QSharedPointer<AbstractNamespaceItem> parentItem = qSharedPointerDynamicCast<AbstractNamespaceItem>(m_parent);
+
+  if ((!m_filter.isEmpty() && m_filter.pattern().compare(m_operations->defaultFilter()) != 0) || parentItem == 0)  {
```

Comment:
It's much better to pass filter to NamespaceItem constructor instead of this "magic" :)